### DISCUSSION TITLE
using pageCount on local pagination need 2 nextPage clic each time

### DISCUSSION THREAD

**Discussion by SpiderSeb** [TanStack/table#3890] [Q&A]
### Describe the bug

When using the pageCount setting (with a value, or even with undefined), we need to clic on the next button twice to have navication on the next page effective.

### Your minimal, reproducible example

https://react-table.tanstack.com/docs/examples/pagination

### Steps to reproduce

On the pagination example
https://react-table.tanstack.com/docs/examples/pagination

add pageCount: undefined in the useTable settings like this :
`useTable(
    {
      columns,
      data,
      initialState: { pageIndex: 2 },
      pageCount: undefined
    },
    usePagination
  )`

then try to navigate on next page. Each time you have to clic twice instead of once.

### Expected behavior

As a user, I expect to have to clic once on the next button to navigate on the next page

### How often does this bug happen?

Every time

### Platform

mac & latest chrome (at least)

### react-table version

7.7.0

### TypeScript version

4.0.3

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by Duckinm** [comment]
My pagination also not work very well, sometimes need to click on it many times.
